src/core/obj.js

```diff
@@ -135,6 +135,19 @@ var Dict = (function DictClosure() {
       return Promise.resolve(value);
     },
 
+    // Same as get(), but dereferences all elements if the result is an Array.
+    getArray: function Dict_getArray(key1, key2, key3) {
+      var value = this.get(key1, key2, key3);
+      if (!isArray(value)) {
+        return value;
+      }
+      var valueLen = value.length, valueArray = new Array(valueLen);
```

Comment:
That is a good point as indeed I thought that we were dealing with a copy here, not a reference to the original dictionary data. In that case we should clearly indicate that this new array is intended as a copy of the original, so using `slice()` will probably help, yes.